### PR TITLE
[5.8] Add Arr::addWhen

### PR DESCRIPTION
**Pull request by phikhi**
### Proposal

This a proposal to add an `Arr::addWhen` function.

This function is a simple wrapper around the existing `Arr::add` function. 
It needs a verified condition as fourth (optional) parameter, to add the specified data to the array. 

### Usage

```php
Arr::addWhen([], 'key', 'value', (1+1 === 2));
// returns ['key' => 'value'];

// if the condition is not provided, it will be true by default.
Arr::addWhen([], 'key', 'value');
// returns ['key' => 'value'];

Arr::addWhen([], 'key', 'value', (1+1 === 3));
// returns [];
```

Or you can use the helper
```php
array_add_when($array, 'key', 'value', (1+1 === 2));
// returns ['key' => 'value'];
```

Hope it can be useful. If not, that's ok too. ;)

### CODE REVIEW

**Comment by phanan** [comment]
:/ What does this save? It's a wrapper around literally one line of code… I know Laravel has similar methods (`abort_if`, `abort_unless` etc.) but those have arguably more usage than this.

**Comment by phikhi** [comment]
@phanan one of those usecases would be this one : 

```php

// some existing array in your code.
$array = [
    'foo' => 'bar',
];

// this could be replaced
if ($thisConditionIsTrue) {
    $array = Arr::add($array, 'key', 'value');
}

// with
$array = Arr::addWhen($array, 'key', 'value', $thisConditionIsTrue);
```

**Comment by ahinkle** [comment]
You should be able to add this as a Macro to your application. No need to add this for the framework to maintain as the use case would be low. It also opens the door for other support helpers to add "when" on everything.

**Comment by phikhi** [comment]
@ahinkle Yes that is what I actually do in some projects. That's why I was proposing it. 
I think i will continue to do it like so.

@phanan @ahinkle Thanks for your opinion 😉

File: src/Illuminate/Support/helpers.php [hunk 1]
```diff
@@ -47,6 +47,24 @@ function array_add($array, $key, $value)
     }
 }
 
+if (!function_exists('array_add_when')) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/helpers.php]: Probably dont add this new function.

**Pull request closed without merging**